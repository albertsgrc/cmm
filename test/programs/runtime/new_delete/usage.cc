/*

    description{Tests new and delete basic functionality}
    
*/

#include <iostream>
using namespace std;

int a[10];

int main() {

    cout << a << endl; // INITIAL + 10*4

    int * x = new int;

    cout << x << endl; // INITIAL + 10*4

    delete x;

    x = new int;

    cout << x << endl; // INITIAL + 10*4

    int * y = new int;

    cout << y << endl; // INITIAL + 10*4 + 4

    int * z = new int[10]; // INITIAL + 10*4 + 4 + 10*4

    cout << z << endl;

    delete[] z;
    delete y;
    delete x;

    x = new int;

    cout << x << endl; // INITIAL + 10*4

    delete x;
}