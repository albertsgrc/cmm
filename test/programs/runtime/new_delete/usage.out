0x80000110
0x80000140
0x80000140
0x80000158
0x80000170
0x80000140
